src/main.c

```diff
@@ -397,146 +397,111 @@ static void apply_operations(numberstack* numbers, operation** current_op) {
-                        in[pos] = '_';
-                    }
-                    else {
-                        // Move the cursor one position back
-
-                        in[pos] = in[pos -1]; // This has to use -1 instead of --, as otherwise it does it in the wrong order and it goes wrong 
-                        in[--pos] = '_';
-                    }
+                    pos--;
+                    browsing = 1; // The left arrow will always be in browsing mode, so no need for a check
                 }
                 searched = 1;
             }
-            else if (inp == 'C')
-            {
+            else if (inp == 'C') {
                 // Right arrow
-                if (browsing) {
-                    // This uses len-2 instead of len-1 because the cursor makes len one longer
-
-                    if (pos < len - 2) {
-                        // If the cursor is still in the middle of the text
-                        
-                        in[pos] = in[pos + 1];
-                        pos++;
-                        in[pos] = '_';
-                    }
-                    else if (pos < len) {
-                        // If the cursor is moving out of the text
-
-                        in[pos] = in[pos + 1]; // Again, ++ doesn't work (see the left arrow)
-
-                        in[++pos] = '\0';
-                        len = pos;
+                if (browsing) {    // The right arrow should only work while in the middle of the input
+                    pos++;
+    
+                    // Exit browsing mode if the cursor is at the end of the input
+                    if (pos == len) {
                         browsing = 0;
                     }
                 }
                 searched = 1;
             }
         }
 
-        if (inp == 127)
-        {
+        if (inp == 127) {
             // Backspace
 
-            if (pos != 0){
+            if (pos != 0) {
                 pos--;
                 len--;
+                inp = '\0';
+            }
+            else {
+                // Skip printing if backspace was pressed but nothing was done, otherwise a strange undefined character is printed
+                searched = 1;
             }
-            inp = '\0';
         }
 
-        if(!searched) {
-            // Prevent user to input more than MAX_IN
-            if (len <= MAX_IN && !browsing) {
+        // Prevent user to input more than MAX_IN
+        if(!searched && len <= MAX_IN) {
+            if (!browsing) {
                 // If the cursor is at the end of the text
                 
                 // Append char to in array
                 in[pos] = inp;
                 in[++pos] = '\0';
                 len++; // Make sure that len is still equal to pos
 
-                if (inp == '\0')
-                {
+                if (inp == '\0') {
                     // Clear screen from previous input
                     mvwprintw(inputwin, 1, 22 + --len ," ");
```

Comment:
And here the same thing, `mvwprintw(inputwin, 1, 22 + --len ," ");` can you add a space after the comma please?

Thank you so much, this is awesome work.

Reply:
I did it through github somehow, ahahah, I though I'd have to send a pull request to your fork but apparently it edited the pull request directly? interesting :)